zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/RestoreFromBackupTool.java

```diff
@@ -274,8 +305,36 @@ private void parseRestoreTempDir(CommandLine cl) {
     }
 
     if (restoreTempDir == null) {
-      // Default address for restore temp dir if not set. It will be deleted after the restoration is done.
-      this.restoreTempDir = new File(snapLog.getDataDir(), "RestoreTempDir_" + zxidToRestore);
+      if (doSpotRestoration) {
+        throw new BackupException(
+            "Local restore temp dir path is not defined for spot restoration.");
+      } else { // For offline restoration
+        // Default address for restore temp dir if not set. It will be deleted after the restoration is done.
+        this.restoreTempDir = new File(snapLog.getDataDir(), "RestoreTempDir_" + zxidToRestore);
```

Comment:
Please clarify the comment: If the user hasn't provided the restore temp dir parameter, then the tool will just create a temporary folder inside `snapLog` and delete it afterwards.